### PR TITLE
Add easy to find information of linked phase archive

### PR DESCRIPTION
**Pull request by amickan**
Addresses https://github.com/comic/grand-challenge.org/issues/3531

This adds a tab to the Challenge admin for each algorithm phase which lists relevant information about the linked archive: 
image

The page then tells the challenge admin which archive is linked to it, which interfaces the archive items need to contain and how many jobs will be scheduled per submission based on the current content of the archive: 

image

Alternatively, it says: 
image

### CODE REVIEW

File: app/tests/evaluation_tests/test_views.py [hunk 1]
```diff
@@ -1668,3 +1668,32 @@ def test_submission_create_sets_limits_correctly_with_predictions(client):
 
     assert submission.algorithm_requires_gpu_type == ""
     assert submission.algorithm_requires_memory_gb == 0
+
+
+@pytest.mark.django_db
+def test_phase_archive_info_permissions(client):
```
Comment (jmsmkn) [inline app/tests/evaluation_tests/test_views.py:1674]: We need to test cross-challenge permissions, that the correct phase is check for permissions for phases with the same slugs but different challenges.

File: app/grandchallenge/evaluation/views/__init__.py [hunk 2]
```diff
@@ -1173,3 +1175,15 @@ def get_success_url(self):
...
             },
         )
+
+
+class PhaseArchiveInfo(
+    LoginRequiredMixin, CachedPhaseMixin, AccessMixin, TemplateView
+):
+    template_name = "evaluation/phase-archive-info.html"
+
+    def dispatch(self, request, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (amickan) [reply app/grandchallenge/evaluation/views/__init__.py:1185]: With `ObjectPermissionRequiredMixin` we'd need to use `get_permission_object` here, and I thought we had agreed that we didn't want to do that anymore, and instead use the `AccessMixin`.

[the remark this answers is not in the document]
Comment (amickan) [reply app/grandchallenge/evaluation/views/__init__.py]: Ok, fair enough. That `DetailView` then needs to override `get_object` and subset the queryset by challenge, or else we run into issues with phases that have the same slug but are from different challenges.

[the remark this answers is not in the document]
Comment (jmsmkn) [reply app/grandchallenge/evaluation/views/__init__.py]: Exactly, that's why I asked about that test.

[the remark this answers is not in the document]
Comment (amickan) [reply app/grandchallenge/evaluation/views/__init__.py]: Yes, I have extended the test.

**Review by jmsmkn — APPROVED** [review]

**Pull request merged**